docs/enhanced-depends-logic.md

```diff
@@ -0,0 +1,74 @@
+# Enhanced Depends Logic
+
+![alpha](assets/alpha.svg)
+
+> v2.8 and after
+
+## Introduction
+
+Previous to version 2.8, the only way to specify dependencies in DAG templates was to use the `dependencies` field and
+specify a list of other tasks the current task depends on. This syntax was limiting because it does not allow the user to
+specify which _result_ of the task to depend on. For example, a task may only be relevant to run if the dependent task
+succeeded (or failed, etc.).
+
+## Depends
+
+To remedy this, there exists a new field called `depends`, which allows users to specify dependent tasks, their statuses,
+as well as any complex boolean logic. The field is a `string` field and the syntax is expression-like with operands having
+form `<task-name>.<task-result>`. Examples include `task-1.Suceeded`, `task-2.Failed`, `task-3.Damenoed`. The full list of
+available task results is as follows:
+
+|  Task Result | Description    |
+|:------------:|----------------|
+| `.Succeeded` | Task Succeeded |
+| `.Failed` | Task Failed |
+| `.Errored` | Task Errored |
+| `.Skipped` | Task Skipped |
+| `.Completed` | Task Succeded or Failed |
+| `.Any` | Task Skipped or Completed |
```

Comment:
this isn't what "any" means?

Reply:
Talked to @jessesuen, will remove this for now.